internal/expr/parser.go

```diff
@@ -343,34 +365,33 @@ func (p *Parser) skipStringLiteral() (bool, error) {
 	return false, nil
 }
 
-// peekByte returns true if the current byte equals the one passed as
-// parameter.
-func (p *Parser) peekByte(b byte) bool {
-	return p.pos < len(p.input) && p.input[p.pos] == b
+// peekChar returns true if the current char equals the one passed as parameter.
+func (p *Parser) peekChar(c rune) bool {
+	return p.pos < len(p.input) && p.char == c
 }
 
-// skipByte jumps over the current byte if it matches the byte passed as a
+// skipChar jumps over the current char if it matches the char passed as a
 // parameter. Returns true in that case, false otherwise.
-func (p *Parser) skipByte(b byte) bool {
-	if p.pos < len(p.input) && p.input[p.pos] == b {
-		p.advanceByte()
+func (p *Parser) skipChar(c rune) bool {
+	if p.pos < len(p.input) && p.char == c {
+		p.advanceChar()
 		return true
 	}
 	return false
 }
 
-// skipByteFind looks for a byte that matches the one passed as parameter and
+// skipCharFind looks for a char that matches the one passed as parameter and
 // then advances the parser to jump over it. In that case returns true. If the
-// end of the string is reached and no matching byte was found, it returns
+// end of the string is reached and no matching char was found, it returns
 // false and it does not change the parser.
-func (p *Parser) skipByteFind(b byte) bool {
+func (p *Parser) skipCharFind(c rune) bool {
```

Comment:
As Go's `rune` type already handles what a char is (not to be confused with C's 1-byte chars), I think it could be nice just to call these runes everywhere to avoid a duplicate concept